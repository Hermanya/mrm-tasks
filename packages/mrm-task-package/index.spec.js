'use strict';

jest.mock('fs');
jest.mock('mrm-core/src/util/log', () => ({
	added: jest.fn(),
}));

const getConfigGetter = require('../../mrm').getConfigGetter;
const vol = require('memfs').vol;
const task = require('./index');

afterEach(() => vol.reset());

it('should add package.json', () => {
	task(
		getConfigGetter({
			name: 'Gendalf',
			url: 'http://middleearth.com',
			github: 'gendalf',
		})
	);

	expect(vol.toJSON()).toMatchSnapshot();
});
