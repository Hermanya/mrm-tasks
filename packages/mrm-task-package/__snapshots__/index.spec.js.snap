// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add package.json 1`] = `
Object {
  "/package.json": "{
  \\"name\\": \\"\\",
  \\"version\\": \\"1.0.0\\",
  \\"description\\": \\"\\",
  \\"author\\": {
    \\"name\\": \\"Gendalf\\",
    \\"url\\": \\"http://middleearth.com\\"
  },
  \\"homepage\\": \\"https://github.com/gendalf/Gendalf\\",
  \\"repository\\": \\"gendalf/Gendalf\\",
  \\"license\\": \\"MIT\\",
  \\"engines\\": {
    \\"node\\": \\">=4\\"
  },
  \\"main\\": \\"index.js\\",
  \\"files\\": [
    \\"index.js\\"
  ],
  \\"scripts\\": {},
  \\"keywords\\": []
}",
}
`;
